python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1481,9 +1485,56 @@ def asset_check_spec(self) -> AssetCheckSpec:
         return self._op_execution_context.asset_check_spec
 
     @public
-    def partition_key_range_for_asset_key(self, asset_key: AssetKey) -> PartitionKeyRange:
-        """TODO - implement in stacked pr."""
-        pass
+    def partition_key_range_for_asset_key(
+        self, asset_key: CoercibleToAssetKey, is_dependency: bool = False
```

Comment:
still not super happy with the `is_dependency` solution to the self dependent asset problem. here's my comment from the first PR exploring this:

--- 
the is_dependency bool param here is to handle the self-dependent partition case. Basically when you have a self-dependent partition, if you said `context.partition_key_for_asset_key("my_asset")` it's unclear if you want the partition key of my_asset as a dependency, or as is being currently materialized. I'm handling this (for now at least) by allowing users to specify `is_dependency` to indicate they want it loaded as a dependency.

In all other cases, the function should be able to figure out the correct partition key to load given the current asset

I'm not set on this solution, so if you have any concerns or other ideas let me know